Implementation of unified EMAC driver for Renesas mbed boards

TomoYamanaka opened the pull request:
### Description
Based on the driver so far, Renesas implemented the emac driver for GR-PEACH and VK-RZ/A1H.

The Greentea test netsocket and emac test passed. Test results are here.

[test_all_GCC_GR-PEACH.txt](https://github.com/ARMmbed/mbed-os/files/1983126/test_all_GCC_GR-PEACH.txt)
[test_emac_GCC_GR-PEACH.txt](https://github.com/ARMmbed/mbed-os/files/1983133/test_emac_GCC_GR-PEACH.txt)


### Pull request type

[ ] Fix  
[x] Refactor  
[ ] New target  
[] Feature  
[ ] Breaking change

kjbracey commented on targets/TARGET_RENESAS/TARGET_RZ_A1XX/ethernet_api.c:
```diff
@@ -315,6 +321,61 @@ void ethernetext_set_link_mode(int32_t link) {
+            tsu_ten_tmp  |= (1 << cnt);
+            break;
+        }
+    }
```
Unlikely to happen, but if you run out of room in the 32-entry table, you should do "set all multicast". You can't leave the caller unable to receive the group they asked for.

kjbracey reviewed (commented):
Code looks good to me, just a little multicast tweak.

kjbracey commented:
`feature-emac` has just been rebased - can you rebase this please?

TomoYamanaka commented:
Ok. I will update my code in based on your above comment and rebase my commit after that.

TomoYamanaka commented:
~@kjbracey-arm~

~Although I rebased my commit to `942209e`, I faced the compile error only IAR.~

TomoYamanaka commented:
~@kjbracey-arm~

~Sorry, I have sent you an unfinished message.~

~Although I rebased my commit to 942209e, I faced the compile error only at IAR.~
~It seem that ARMCC and GCC is no problem.~
~The error message is here:~


~Compiling was successful if it was the environment I sended PR (before rebase).~
~Could you check the problem ?~

~The command that I run is here:~
~`mbed test --compile --run -m <target> -t IAR -v -n tests-network-emac --app-config TESTS/network/emac/template_mbed_app.txt`~

TomoYamanaka commented:
@kjbracey-arm

Sorry to bother you again.

It is seems that the above phenomenon occured by a mistake of my environment and test order.
I will withdraw the my 2 comments.

TomoYamanaka committed:
Implementation of unified EMAC driver for Renesas mbed boards

Implementation of unified EMAC driver for Renesas mbed boards

Based on the driver so far, Renesas implemented the emac driver for GR-PEACH and VK-RZ/A1H.
The mainly changes is below.
- Add the connection part with LWIP according to the unified emac specification.
- Add three new multicast functions(add, remove, set_all).

The Greentea test netsocket and emac test passed.

TomoYamanaka commented:
I rebased my commit with reflected the above comment regarding multicast.
